library/init/meta/interactive.lean

```diff
@@ -272,51 +258,57 @@ match e with
 | _                     := i_to_expr p
 end
 
-private meta def to_symm_expr_list : list (pexpr × pos) → tactic (list (bool × expr × pos))
-| []             := return []
-| ((p, pos)::ps) :=
+private meta def maybe_save_info : option pos → tactic unit
+| (some p) := save_info p.1 p.2
+| none     := skip
+
+private meta def symm_expr := bool × expr × option pos
+
+private meta def to_symm_expr_list : list pexpr → tactic (list symm_expr)
+| []      := return []
+| (p::ps) :=
   match is_neg p with
-  | some a :=  do r ← to_expr' a, rs ← to_symm_expr_list ps, return ((tt, r, pos) :: rs)
-  | none   :=  do r ← to_expr' p, rs ← to_symm_expr_list ps, return ((ff, r, pos) :: rs)
+  | some a :=  do r ← to_expr' a, rs ← to_symm_expr_list ps, return ((tt, r, pexpr.pos p) :: rs)
```

Comment:
Nice. 
BTW, this is the position we use to save the intermediate states of `rw`.